sklearn/metrics/_regression.py

```diff
@@ -986,3 +987,107 @@ def mean_gamma_deviance(y_true, y_pred, *, sample_weight=None):
     1.0568...
     """
     return mean_tweedie_deviance(y_true, y_pred, sample_weight=sample_weight, power=2)
+
+
+def d2_tweedie_score(y_true, y_pred, *, sample_weight=None, power=0):
+    """D^2 regression score function, percentage of Tweedie deviance explained.
+
+    Best possible score is 1.0 and it can be negative (because the model can be
+    arbitrarily worse). A model that always predicts a constant value for the expected
+    value of y, disregarding the input features, would get a D^2 score of 0.0.
```

Comment:
What do you mean by "the expected value of y"? Any constant value would work, or does it mean the mean on y_true?

Reply:
Yes, it means a model that constantly uses the empirical mean of the observed y as prediction. I thought this one would be even clearer than the sentence for R2. I'll make it more precise.